### PR TITLE
[imgui] Update to 1.90

### PR DESCRIPTION
**Pull request by moritz-h**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/imgui/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "imgui",
-  "version": "1.89.9",
+  "version": "1.90.0",
```
Comment (Adela0814) [inline ports/imgui/vcpkg.json:3]:
```suggestion
  "version": "1.90",
```

File: ports/imgui/portfile.cmake [hunk 2]
```diff
@@ -4,16 +4,16 @@ if ("docking-experimental" IN_LIST FEATURES)
...
+       SHA512 50953097ff809bca0e1cee55268eaa8dad001cf8fae85ae7623f484752d16bdf8e63c04227786361f27e73f7aeda5a8d1f9b0cee83c2aca108ce2712a21b74d9
        HEAD_REF docking
        )
 else()
     vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO ocornut/imgui
-    REF v${VERSION}
-    SHA512 42021b06b611b58222b09fab8db2c34e992c3dc4fbaa175e09833c66c90d04b4a4e7def16a732535335c0ac5ff014d235835511a5d9a76d32b4395b302146919
+    REF "v1.90"
```
Comment (Adela0814) [inline ports/imgui/portfile.cmake:15]:
```suggestion
    REF "v${VERSION}"
```

File: ports/imgui/portfile.cmake [hunk 3]
```diff
@@ -4,16 +4,16 @@ if ("docking-experimental" IN_LIST FEATURES)
     vcpkg_from_github(
        OUT_SOURCE_PATH SOURCE_PATH
        REPO ocornut/imgui
-       REF a1b60fc1f5589d498ab1080c2572da725fcbd0e3
-       SHA512 ac6117f6adf9418af3a2db5392f1316be50a94c38e78cd1eadc0e0a71dfbb5536507aaf4d9d3b3468b5a30ebf143b806e2774f274e2098e1217ed93080fe81c7
+       REF "v1.90-docking"
```
Comment (Adela0814) [inline ports/imgui/portfile.cmake:7]:
```suggestion
       REF "v${VERSION}-docking"
```

**Review by Adela0814 — DISMISSED** [review]

**Comment by Adela0814** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Review by PhoebeHui — APPROVED** [review]
LGTM

**Pull request merged**